pkg/publish/options.go

```diff
@@ -42,38 +38,11 @@ func WithUserAgent(ua string) Option {
 	}
 }
 
-// WithAuth is a functional option for overriding the default authenticator
-// on a default publisher.
-func WithAuth(auth authn.Authenticator) Option {
-	return func(i *defaultOpener) error {
-		i.auth = auth
-		return nil
-	}
-}
-
-// WithAuthFromKeychain is a functional option for overriding the default
+// WithKeyChain is a functional option for overriding the default
 // authenticator on a default publisher using an authn.Keychain
-func WithAuthFromKeychain(keys authn.Keychain) Option {
```

Comment:
We shouldn't rename this, since other users who use ko as a library might use it. Same with `WithAuth` above.